bots/s3_block_public_access.py

```diff
@@ -0,0 +1,122 @@
+"""
+## s3_block_public_access
+What it does: turn on S3 Bucket Block public access
+Usage: AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=<true|false>  BlockPublicPolicy_NewPUTRequests=<true|false>
+Limitations: none
+Notes:
+    -  before running this bot, ensure that your applications will work correctly without public access
+    - works with JSON - Full / Basic entity findings
+Examples:
+
+    Block public access to buckets and objects granted through new public ACLs and Bucket Policies:
+    AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=true  BlockPublicPolicy_NewPUTRequests=true
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies
+    ( granted through new public ACLs and Bucket Policies Configuration Stays the same ):
+    AUTO: s3_block_public_access
+
+    Block public access to buckets and objects granted through any existing ACLs and Bucket Policies But not for
+    new public ACLs and Bucket Policies:
+    AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=false  BlockPublicPolicy_NewPUTRequests=false
+
+"""
+
+import boto3
+from botocore.exceptions import ClientError
+
+usage = 'AUTO: s3_block_public_access BlockPublicAcls_NewPUTRequests=<true|false>  BlockPublicPolicy_NewPUTRequests=<true|false> \n '
```

Comment:
Why usage is part of the function ?

Reply:
Got the idea from another bot (sg_single_rule_delete), it's easier to return the usage in the output message in case of mishandling the bot parameters